File: leetcode/AIcoding/4.GPT-4o/c_solutions/3048-earliest-second-to-mark-indices-i.c
int earliestFullMark(int* marks, int marksSize) {
    int count[marksSize + 1];
    memset(count, 0, sizeof(count));

    for (int i = 0; i < marksSize; i++) {
        count[marks[i]]++;
    }

    for (int i = 0; i < marksSize + 1; i++) {
        if (count[i] == 0) {
            return i;
        }
    }

    return -1; // This line should never be reached
}

File: leetcode/AIcoding/4.GPT-4o/go_solutions/3048-earliest-second-to-mark-indices-i.go
func earliestFullMark(times []int) int {
    n := len(times)
    if n == 0 {
        return -1
    }

    marks := make(map[int]bool)
    for i := 0; i < n; i++ {
        marks[times[i]] = true
    }

    for i := 0; i <= 100; i++ {
        if !marks[i] {
            return i
        }
    }

    return -1
}